au4/src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -109,6 +109,13 @@ bool Au3Interaction::canPasteClips(const std::vector<TrackId>& dstTracksIds, sec
             //! TODO AU4: show dialog
             return false;
         }
+
+        if (dstWaveTrack->NChannels() == 1 && data[i].track.get()->NChannels() == 2) {
+            interactive()->error(muse::trc("message", "Message"), std::string(
+                "The content you are trying to paste will span across more tracks than you"
+                " currently have available. Add more tracks and try again."));
+            return false;
```

Comment:
It is better not to use interactivity in services without extreme necessity.
It is better to return Ret, and somewhere higher, at the ViewModel level, show a dialog to the user...
Because this method (like other service methods) can be used in different scenarios, some of which may not involve a dialog...
Or simply, if we want to write a unit test - this dialog will be a problem